tests/cases/compiler/conditionalTypeExtendsNewline.ts

```diff
@@ -0,0 +1,29 @@
+// Repro from #53589
+
+type Example1 = string 
+     extends string ? true : false;
```

Comment:
according to [@RyanCavanaugh 's comment ](https://github.com/microsoft/TypeScript/issues/53589#issuecomment-1490572446) this should be an error, but it's not anymore.

Reply:
I thought that was implying that it's a bug that we error on it now?

Reply:
(The PR looks simple enough that it feels like it's going to be harder to disallow certain constructs...)

Reply:
My thought process was that the `extends` now behaves similarly to how the ternary is parsed.  I.e. 
```
const a = true
       ? 'foo' : 'bar';
```
Is parsed fine and the `extends` here feels very similar to that.  Additionally, in the PR where the new line check was added  (#21649) it was not added to prevent having _any_ `extends` after a newliine, but specifically to counter the case where that `extends` is actually a key which is still handled here.

Reply:
Ryan's words were:

> Some of this is working as intended (X5 is a working-as-intended ASI for example) but this one is a bug

```ts
type M<T> = {
    [K in keyof T as K
    extends "" ? K : K]: any;
}
```

X5 was this:

```ts
type X5<T> = T
extends number ? true : false;
```

I wrote a whole set of examples that assumed that `extends` was only a keyword in certain contexts, but that's wrong. It's a reserved word; I don't know why it's allowed as a class property name.

So: 

1. X5 should actually work, Ryan was wrong.
2. I don't know all the contexts `extends` can be used as an identifier. Is it just class properties and property access expressions? What can follow a class property? `: ? = ;` plus newline are the ones that come to mind.

Reply:
I see, I missed that `X5` reference.

That being said, given the idea of "extends is the ternary of type space", I'm personally pretty amenable to it behaving as closely to a ternary operator as possible.

Reply:
Wouldn't it be more of a quaternary? I feel like you can either see it as `A extends B ? C : D` or as `A extends B` producing the boolean of type space that can be passed into the ternary of type space. But maybe that doesn't change the arguments either way.

Reply:
I guess if we're being pedantic, then yes, I should have said "conditional expression". Not really the point.

Reply:
But I guess my argument is pointless given `A extends B` is the "condition`, so `extends` is not the `?`. In which case, whatever behavior people want is fine with me.

Reply:
Yeah, that was my confusion. I think `A extends B` is more comparable to `A in B` in JavaScript, e.g. `A in B ? C : D`. But that's more of gut feeling than anything based on a deep understanding of either operator.

Reply:
Sorry for the delay on this but I have updated the check to more robustly determine if extends could be a instance variable in a class and added some more test cases to exercise this case.